core_java/twitter/src/main/java/ca/jrvs/apps/twitter/controller/TwitterController.java

```diff
@@ -0,0 +1,90 @@
+package ca.jrvs.apps.twitter.controller;
+
+import ca.jrvs.apps.twitter.model.Tweet;
+import ca.jrvs.apps.twitter.service.Service;
+import ca.jrvs.apps.twitter.util.CreateTweetUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+@org.springframework.stereotype.Controller
+public class TwitterController implements Controller{
+
+    private static final String COORD_SEP = ":";
+    private static final String COMMA = ",";
+    private Service service;
+
+    @Autowired
+    public TwitterController(Service service){
+        this.service = service;
+    }
+
+    /**
+     * Parse user argument and post a tweet by calling service classes
+     *
+     * @param args
+     * @return a posted tweet
+     * @throws IllegalArgumentException if args are invalid
+     */
+    @Override
+    public Tweet postTweet(String[] args) {
+        if (args.length != 3){
+            throw new IllegalArgumentException("Invalid Number of Arguments");
+        }
+
+        String text = args[1];
+        String coords = args[2];
+        String[] corr = coords.split(COORD_SEP);
+
+        if (corr.length < 2){
+            throw new IllegalArgumentException("Invalid Number of Coordinates for tweet");
+        }
+
+        Double lng;
+        Double lat;
+
+        try{
+            lng = Double.parseDouble(corr[0]);
+            lat = Double.parseDouble(corr[1]);
+        } catch (NumberFormatException e){
+            throw new IllegalArgumentException("Invalid Input: Error parsing coordinates");
```

Comment:
throw new IllegalArgumentException("Invalid Input: Error parsing coordinates", e);

make sure you pass the caught exception. otherwise you will lose the stacktrace